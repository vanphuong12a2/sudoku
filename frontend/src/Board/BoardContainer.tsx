import React from 'react';
import Board from './Board';

interface Props {
    boardData: (number | undefined)[][]
}

interface State {
    currentBoard: (number | undefined)[][]
}

class BoardContainer extends React.Component<Props, State>{

    constructor(props: Props) {
        super(props);
        this.state = {currentBoard: this.props.boardData};
    }

    render() {
        return (
            <Board
                boardData={this.props.boardData}
                currentBoard={this.state.currentBoard}/>
        )
    }
}

export default BoardContainer;