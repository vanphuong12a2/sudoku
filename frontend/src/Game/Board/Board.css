.main-board {
    margin:1em;
    max-width: 450px;
    max-height: 450px;
    width: 100%;
    position: relative;
}

.main-board td {
    border:1px solid;
    padding: 0;
}
.main-board td:first-child {
    border-left:solid;
}
.main-board td:nth-child(3n) {
    border-right:solid ;
}
.main-board tr:first-child {
    border-top:solid;
}
.main-board tr:nth-child(3n) td {
    border-bottom:solid ;
}

.main-board caption {
    text-align: left;
}

.main-board input,
.main-board input:hover,
#root .main-board input:focus {
    width: 50px;
    height: 50px;
    text-align:center;
    border: 0;
    margin: 0;
    font-size: 1.7rem;
}

.main-board input.writable {
    color: green;
}

.main-board input.readonly {
    cursor: pointer;
    color: black;
}

.main-board input.invalid {
    background: pink;
    border: 0;
    box-shadow: none;
}

.main-board i {
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    font-size: 3em;
}

