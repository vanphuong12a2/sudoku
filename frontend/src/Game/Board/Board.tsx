import React from 'react';
import Cell, {EMPTY_CELL} from '../Cell/Cell';
import './Board.css';

interface Props {
    loading: boolean
    boardData: number[][]
    currentBoard: number[][]
    onCellChange: (rowIndex: number, cellIndex: number) => (newValue: number) => void
}

const Board = (props: Props) => {

    const boardTable = <table>
        <tbody>
        {props.currentBoard.map((row, rowIndex) => {
            return (
                <tr key={rowIndex}>
                    {
                        props.currentBoard[rowIndex].map((cellData, cellIndex) => {
                            return (
                                <Cell
                                    key={cellIndex}
                                    cellData={cellData}
                                    readOnly={props.boardData[rowIndex][cellIndex] !== EMPTY_CELL}
                                    onCellChange={props.onCellChange(rowIndex, cellIndex)}
                                />
                            );
                        })
                    }
                </tr>);
        })}
        </tbody>
    </table>;

    const loadingSpinner = <i className='fa fa-spinner fa-spin'/>;

    return (
        <div className='main-board'>
            {props.loading ? loadingSpinner : boardTable}
        </div>
    );
};

export default Board;