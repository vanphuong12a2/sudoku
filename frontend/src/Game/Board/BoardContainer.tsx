import React from 'react';
import Board from './Board';

interface Props {
    boardData: (number)[][]
}

interface State {
    currentBoard: (number)[][]
}

class BoardContainer extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
        this.state = {currentBoard: this.copyBoard(this.props.boardData)};
    }

    private copyBoard = (board: number[][]) => {
        return board.map(row => row.slice());
    };

    componentWillReceiveProps(nextProps: Props) {
        if (nextProps.boardData !== this.props.boardData) {
            this.setState({currentBoard: this.copyBoard(nextProps.boardData)});
        }
    }

    render() {
        return (
            <Board
                boardData={this.props.boardData}
                currentBoard={this.state.currentBoard}
                onCellChange={this.onCellChange}
            />
        )
    }

    public onCellChange = (rowIndex: number, cellIndex: number) => (newValue: number) => {
        this.state.currentBoard[rowIndex][cellIndex] = newValue;
        this.setState({currentBoard: this.state.currentBoard});
    }
}

export default BoardContainer;