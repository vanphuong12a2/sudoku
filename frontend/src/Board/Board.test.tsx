import React from 'react';
import {create} from 'react-test-renderer';
import Board from './Board';

describe('<Board />', () => {

    describe('Snapshots', () => {

        it('renders boardData correctly', () => {
            let boardData = [
                [undefined,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,undefined,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,undefined,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9]
            ];
            let currentBoard = [
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9],
                [1,2,3,4,5,6,7,8,9]
            ];
            const board = create(<Board boardData={boardData} currentBoard={currentBoard}/>);
            expect(board.toJSON()).toMatchSnapshot();
        });
    });

});