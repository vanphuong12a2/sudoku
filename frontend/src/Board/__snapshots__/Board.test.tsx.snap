// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> Snapshots renders boardData correctly 1`] = `
<table
  className="main-board"
>
  <tbody>
    <tr>
      <td>
        <input
          className=""
          maxLength={1}
          readOnly={false}
          value={1}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={2}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={3}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={4}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={5}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={6}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={7}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={8}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={9}
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={1}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={2}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={3}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={4}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={5}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={6}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={7}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={8}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={9}
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={1}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={2}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={3}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={4}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={5}
        />
      </td>
      <td>
        <input
          className=""
          maxLength={1}
          readOnly={false}
          value={6}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={7}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={8}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={9}
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={1}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={2}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={3}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={4}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={5}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={6}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={7}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={8}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={9}
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={1}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={2}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={3}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={4}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={5}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={6}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={7}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={8}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={9}
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={1}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={2}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={3}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={4}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={5}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={6}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={7}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={8}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={9}
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={1}
        />
      </td>
      <td>
        <input
          className=""
          maxLength={1}
          readOnly={false}
          value={2}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={3}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={4}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={5}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={6}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={7}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={8}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={9}
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={1}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={2}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={3}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={4}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={5}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={6}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={7}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={8}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={9}
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={1}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={2}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={3}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={4}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={5}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={6}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={7}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={8}
        />
      </td>
      <td>
        <input
          className="readonly"
          maxLength={1}
          readOnly={true}
          value={9}
        />
      </td>
    </tr>
  </tbody>
</table>
`;
